extensions/grpc/runtime/src/main/java/io/quarkus/grpc/auth/GrpcSecurityRecorder.java

```diff
@@ -43,4 +47,17 @@ public void initGrpcSecurityInterceptor(Map<String, List<String>> serviceClassTo
         container.beanInstance(GrpcSecurityInterceptor.class).init(svcToMethods);
     }
 
+    public void validateSecurityEventsDisabled(String observedSecurityEvent) {
+        boolean securityEventsEnabled = ConfigProvider
+                .getConfig()
+                .getOptionalValue("quarkus.security.events.enabled", boolean.class)
+                .orElse(Boolean.TRUE);
+        if (securityEventsEnabled) {
+            throw new ConfigurationException("""
+                    Found observer method for event type '%s', but the gRPC extension does not support security
+                    events. Either disable security events with the 'quarkus.security.events.enabled'
```

Comment:
What if both GRPC and vertx-http are used ?

Reply:
Considering this is only time where I can reproduce it (not gRPC running on a separate server) and considering gRPC is just a route handler answer is: if gRPC extension is present, security events are not supported.

I'd not make fuss about it, it is temporary and it is done for a very good reason, something super strange and hard to debug is happening there if you fire asynchronous events that activate CDI request context and no, it is not in how I fire events because all this PR does is `event.fireAsync`, you can't make mistake in that.

Reply:
> But if I want to observe OIDC events and also need to use GRPC, failing the builds looks unnecessary to me.

but in case gRPC is just a route handler and Vert.x HTTP security handlers like HTTP Authenticator and HTTP Authorizer are run, you don't know what handler is applied next. How do you know it is going to be REST stack and not a gRPC? I'm also not sure why should applications combine gRPC and REST stacks, wouldn't microservice do one thing?

> What do you mean when you say a big refactoring would be needed, I thought, in case of GRPC, for the events to work there would have to be some GRPC specific code to gave then fired

Problem is not a gRPC specific code firing events, problem is that when HTTP Authenticator and HTTP Authorizer fires events asynchronously it mess with how `io.quarkus.grpc.runtime.supports.context.GrpcRequestContextGrpcInterceptor` works with a context.

> , but this code is not even there so where is a problem if only a warning is issued ?

Code is there and logic that determines inside Vert.x HTTP routes what router will be next is complex.

Reply:
@michalvavrik OK, lets leave the GRPC issue for now